### ISSUE TITLE
Public Dashboards: Internal Server Error on public dashboards

### ISSUE THREAD

**Issue by l-hah** [grafana/grafana#73803]
### What happened?

When sharing a dashboard as a public dashboard, all panels show the "Internal Server Error". However when the dashboard is displayed in normal view, it works without problems.

Issue on the public dashboard: ![grafik](https://github.com/grafana/grafana/assets/139213807/308f82ac-4720-4a98-951c-c36e5398c5ee)
Same dashboard in the standard view 
![grafik](https://github.com/grafana/grafana/assets/139213807/c824c977-7934-4dc3-932e-89c85f5239c5)

### What did you expect to happen?

I expected the dashboard to be displayed just as it is in normal view.

### Did this work before?

Better than now. Previously (v10.0.4, 71b69f9a5e), the panel still showed both the hidden, and active queries, which I thought was fixed in Issue #71326. I cannot tell if this issue still persists in v10.1.0, as no data is displayed at all.

### How do we reproduce it?

1: Create a Grafana Dashboard
2: Add a data source
3: Create a panel in the Dashboard
4: Select any type of visualization (Gauge in my case)
5: Add a query to select some data from the data source

now the panels should produce the error message

### Is the bug inside a dashboard panel?

No. It is an issue with the whole dashboard.

### Environment (with versions)?

Grafana: Grafana v10.1.0 (838218ba20)
OS: Windows 10
Browser: Firefox and Chrome

### Grafana platform?

A downloaded binary

### Datasource(s)?

Microsoft SQL Server

**Comment by usmangt** [comment]
:Hi @l-hah,

Thanks for opening this issue.

Are you using any Dashboard aka template variables in your query?

If yes, then it might be the case as currently:

> Template variables are not supported
> This public dashboard may not work since it uses template variables

If you are not using any template variables but still get it, can you please test it with other Data sources (not just MySQL) and tell if you see any different error message?

Also, please send us the logs. The backend logs (the default location in Linux is `/var/log/grafana/grafana.log`) so that we can have more information about the error message?

**Comment by l-hah** [comment]
I apologize for opening this issue in the first place. There was a mistake on my part. I executed Grafana 10.1.0, imported my test-panel and datasource, and it didn't work. After some major troubleshooting (and reading through the logs, which I didn't know existed), it turned out that I forgot to change the data source from the panels in the dashboard to the new datasource. That might have happened, because the old connector from my live installation, and the connector on my test installation are called the same, but Grafana treats them different, and you need to re-select the datasource on the panels, even if they are called the same.
I'm ashamed at myself that I made this mistake and wasted your time. I'm truly sorry.
On the good news, so far, I'm very happy with the new Grafana 10.1.0 version.

**Comment by usmangt** [comment]
No worries @l-hah 

We are always happy to receive any feedback to improve the product and glad to hear that you resolved it on your own but also happy with the improvements made in the newer version. :slightly_smiling_face: 

I will close this issue :+1:

**Comment by ckanthprojects** [comment]
can you tell me how to resolve it

**Comment by l-hah** [comment]
@ckanthprojects the issue for me was (from what I memember), was that the data source was not selected correctly. Try assigning that panel a different data source, and then switching back to your intended data source. In hindsight, that sounds like total crap, but I remember that this is how I solved it. If that doesn't solve it for you, then I'm sorry. It has been a while since I've resolved that issue, and I know that it was because I missed something or messed something up during the transistion of the dashboard to my test grafana instance.
